test/e2e/metrics_test.go

```diff
@@ -256,14 +258,23 @@ var _ = Describe("Metrics", func() {
 				gauges = append(gauges, gauge)
 				builder.WithMetric(gauge)
 			}
-			sendMetrics(context.Background(), builder.Build(), otlpPushURL)
+			Expect(sendMetrics(context.Background(), builder.Build(), urls.otlpPush())).To(Succeed())
+			Expect(sendMetrics(context.Background(), builder.Build(), urls.otlpPushAt(1))).To(Succeed())
+
+			Eventually(func(g Gomega) {
+				resp, err := httpsAuthProvider.Get(urls.mockBackendExport())
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				g.Expect(resp).To(HaveHTTPBody(SatisfyAll(
+					HaveNumberOfMetrics(len(gauges)))))
```

Comment:
Since we have two backends now, we can also validate the actual metrics (`HaveMetrics`) instead of the pure number. Same for tracing.

Reply:
That is a sharp catch! I extended the validation.